### ISSUE TITLE
Upload doesn't work via browser on iphone

### ISSUE THREAD

**Issue by DavHau** [ente-io/ente#4083]
### Description

One of the killer features of ente.io is the ability to create collaborative albums where my friends can upload media after a group trip, without creating an account or downloading an app, simply by following the shared link.

Though several of my friends who use a modern iPhone cannot upload pictures or videos via the collaborative link.
I am using the official ente hosted service (no self hosting). The upload progress is simply stuck for iPhones using Chrome or Safari.

Please fix this. This is important. I'd really prefer not having to go back to some proprietary service.

### What product are you using?

Ente Photos

### What platform are you using?

Web - Safari

**Comment by mnvr** [comment]
Thank you for the report. Could you expand on the "upload progress is simply stuck"? 

I ask because this issue is not reproducible. Perhaps you're talking of something else than what I'm trying to reproduce. There is some delay on the add button in the native gallery if selecting 50+ images as the Apple Photos app prepares the photos for handoff, maybe you talking of that delay.

Let's consider an example upload. 

1. Open a shared link, and select "Add photos". iPhone prompts for some options, select the "Photo Library" option.
2. This opens the native library. Select some photos. Press the "Add" button. 
3. Once the native library prepare the photos (downloading them from iCloud if needed), the control goes back to the Ente shared link app in the browser, and you should be seeing a upload progress.
4. If the browser is moved to the background when the upload is in progress, the upload will get paused. However, it should resume when the browser comes back again into the foreground.

It would be helpful to know (a) which particular step 2/3/4 is getting stuck, (b) how many photos were being uploaded.

If the upload is getting stuck in step 3 (when the app is actually uploading), it would also be helpful to know if (c) what is the value for the format that was selected when uploading.

<img src="https://github.com/user-attachments/assets/62fcfdd0-abf7-44a9-893f-9164fcd4d3fb" width=400>

**Comment by NoahAndrews** [comment]
My friend just encountered this issue, so I found this thread and asked her to answer your questions. Here's her response: 

> a) It seems to get stuck on step three, where after I click “add” it seems like maybe some photos are downloading from iCloud, but nothing seems to happen and it doesn’t return to the Entre shared link app. When some things do seem to move past this step to the file upload progress, sometimes the web page reloads during the progress and resets itself, so the file doesn’t upload
>
> b) I originally tried to upload 68 photos (a portion were videos), but that seemed to be too much, so it worked when I added about 25 photos only at a time. Once I tried to add some photos and some videos, it didn’t work anymore. I successfully added a short 10 second video, but I tried to add a 3 minute 30 second video, and it seemed to download it from iCloud and bring me back to the shared folder to upload, but after a few seconds the whole folder reloaded on its own before the upload finished and it didn’t add at all

**Comment by mnvr** [comment]
Thank you, this is helpful. So it seems that the issues arises when trying to upload items (either many at once, or larger videos) that are not on the device. In which case Apple Photos downloads them from iCloud, and during this process, Safari reloads the page (likely due to memory pressure on the phone because of the ongoing iCloud downloads).

The browser hands off to the app only after the download from iCloud has finished, so there isn't a trivial way to fix this - the app just isn't in control when this breakage happens. However, maybe there is an alternative way to tackling this, we'll continue to explore.

That said, there just might be limits to how much we can optimize the uploads via browser on the phone, which is why in parallel we also are working on allowing folks to upload to public albums using the app, where we can give a much more reliable experience.

**Comment by NoahAndrews** [comment]
Yeah, the fact that it wasn't possible for her to use the app to do the upload was a little surprising to me (I suggested she download the app from the get-go, but of course it just directed her to the website).
